### PR TITLE
Fuzz template 4125 v4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4125

Describe changes:
- fuzz: enable template protocols
- dns : wrap with `HAVE_LUA` some code
- fuzz : use parsed rules in sigpcap target (as already fixed for aware target)
- detect: do not upgrade base64 decode when fuzzing
- detect: logs an error if a protocol is disabled 

Maybe this PR can be split in different ones.
Like the ` detect: logs an error if a protocol is disabled `, which could then have a different ticket as well ?

Replaces #6694
- fixes the bug found by CIFuzz in base64 (over allocation with a dummy signature)
- fixes the unit tests by having a special case for HTTP_ANY signatures

suricata-verify-pr: 602
https://github.com/OISF/suricata-verify/pull/602
The global point is to increase fuzzing coverage

### CODE REVIEW

File: src/detect-dns-query.c [hunk 1]
```diff
@@ -199,6 +198,23 @@ static int PrefilterMpmDnsQueryRegister(DetectEngineCtx *de_ctx,
...
+}
+
+static int DetectEngineInspectDnsResponse(DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+        const struct DetectEngineAppInspectionEngine_ *engine, const Signature *s, Flow *f,
+        uint8_t flags, void *alstate, void *txv, uint64_t tx_id)
+{
+    return DetectEngineInspectGenericList(
+            de_ctx, det_ctx, s, engine->smd, f, flags, alstate, txv, tx_id);
+}
+#endif
```
Comment (jasonish) [inline src/detect-dns-query.c:217]: These are only used for Lua scripting then?

Comment (catenacyber) [reply src/detect-dns-query.c:217]: Indeed.
See also below previous comment :

> register these generic engines from here for now

**Comment by suricata-qa** [comment]
Information:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.memuse | 541517248 | 505916608 | 107.04% | 

Pipeline 5303

**Comment by catenacyber** [comment]
> fuzz : use parsed rules in sigpcap target (as already fixed for aware target)

By the way, this was what prevented fuzzing to find https://redmine.openinfosecfoundation.org/issues/4503

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/6717

**Pull request closed without merging**